.github/workflows/rust-prs.yml

```diff
@@ -0,0 +1,459 @@
+name: rust-prs
+
+on:
+  workflow_dispatch:
+  pull_request:
+
+env:
+  GRAFBASE_SKIP_ASSET_VERSION_CHECK: 'true'
+  CARGO_INCREMENTAL: 0
+  CARGO_PROFILE_DEV_DEBUG: 0
+  CARGO_PROFILE_TEST_DEBUG: 0
+  CARGO_TERM_COLOR: 'always'
+  DO_NOT_TRACK: 1
+  GRAFBASE_RUDDERSTACK_WRITE_KEY: ${{ secrets.GRAFBASE_RUDDERSTACK_WRITE_KEY }}
+  GRAFBASE_RUDDERSTACK_DATAPLANE_URL: ${{ secrets.GRAFBASE_RUDDERSTACK_DATAPLANE_URL }}
+  RUST_BACKTRACE: 1
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-rust-pr
+  cancel-in-progress: true
+
+jobs:
+  what-changed:
+    runs-on: ubicloud-standard-2
+    outputs:
+      rust: ${{ steps.what-rust-changed.outputs.rust }}
+      grafbase-docker: ${{ steps.paths-changed.outputs.grafbase-docker }}
+      gateway-docker: ${{ steps.paths-changed.outputs.gateway-docker }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          # TODO: Figure out how to not make this zero
+          # Seems like if it's not zero you dont get branches
+          fetch-depth: 0
+
+      - name: Extract the Rust version to use from the `rust-toolchain.toml` file
+        shell: bash
+        run: |
+          rust_version=$(grep "channel" rust-toolchain.toml | cut -d "=" -f 2 | cut -d " " -f 2)
+          echo "RUST_VERSION=$rust_version" >> $GITHUB_ENV
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_VERSION }}
+          components: clippy, rustfmt
+
+      - name: Install what-rust-changed
+        shell: bash
+        run: cargo install --locked --git https://github.com/grafbase/what-rust-changed.git --branch obmarg/qmtxorwqlrtl
```

Comment:
Could we use bininstall here? That would avoid compiling the tool.

Reply:
Once https://github.com/grafbase/what-rust-changed/pull/2 is merged I'm going to tag a release of what-rust-changed.  There's a workflow in that PR that'll publish release binaries and then I'll change this to fetch those.

Don't neccesarily want to use `cargo binstall` - both because I don't want to publish to crates.io _and_ because I saw some weird issues where `cargo binstall` occasionally failed to install itself and just retried endlessly until the build timed out 👎

Reply:
(Weirdly I only saw that cargo binstall issue in one place I was using it, which I've now removed - the binstall of `cargo component` always seemed to work)